lib/fs.ts

```diff
@@ -192,7 +192,7 @@ class FSPath {
 			return;
 		}
 
-		this.writeUpdateNow(dataFetcher, options);
+		return this.writeUpdateNow(dataFetcher, options);
 	}
 	finishUpdate() {
```

Comment:
There's a dangling Promise in this function, by the way. Apparently TSLint doesn't recognize `Promise<void> | void` as a Promise.

Reply:
I return `Promise<T> | T` in a few places in the Streams API for perf reasons; in theory it's not hard to `await` or Promise.resolve` the result if you want a guaranteed Promise.